plugin/src/main/java/com/denizenscript/denizen/scripts/commands/entity/AgeCommand.java

```diff
@@ -57,64 +59,37 @@ public AgeCommand() {
     // - age <player.location.find_entities.within[20]> adult
     // -->
 
-    private enum AgeType {ADULT, BABY}
+    private enum AgeType { ADULT, BABY }
 
-    @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (!scriptEntry.hasObject("entities")
-                    && arg.matchesArgumentList(EntityTag.class)) {
-                scriptEntry.addObject("entities", arg.asType(ListTag.class).filter(EntityTag.class, scriptEntry));
-            }
-            else if (!scriptEntry.hasObject("agetype")
-                    && arg.matchesEnum(AgeType.class)) {
-                scriptEntry.addObject("agetype", AgeType.valueOf(arg.getValue().toUpperCase()));
-            }
-            else if (!scriptEntry.hasObject("age")
-                    && arg.matchesInteger()) {
-                scriptEntry.addObject("age", arg.asElement());
-            }
-            else if (!scriptEntry.hasObject("lock")
-                    && arg.matches("lock")) {
-                scriptEntry.addObject("lock", new ElementTag(true));
-            }
-            else {
-                arg.reportUnhandled();
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("entities") @ArgLinear ObjectTag entities,
+                                   @ArgName("age") @ArgLinear @ArgDefaultText("1") ObjectTag age,
+                                   @ArgName("lock") boolean shouldLock) {
+        if (age instanceof ListTag || age instanceof EntityTag) { // Compensate for legacy age/entity out-of-order support

```

Comment:
This won't work with manual lists (I.e. `1|2|3`) - generally it's easier to cover everything by checking the argument with the most constrains, in this `age` as it can only be an enum or a number.
I'd do a loose "is `entities` valid" check (probably just an `instanceof`), and if not check if it's a valid age and swap; see also the `look` command for a similar example.

Reply:
should be alright now, did some testing and works fine for all cases
also ready to review